plan/physical_plan_builder.go

```diff
@@ -75,8 +77,21 @@ func getRowCountByIndexRange(table *statistics.Table, indexRange *IndexRange, in
 			return 0, errors.Trace(err)
 		}
 		count = count / float64(table.Count) * float64(rowCount)
+		// If the condition is a = 1, b = 1, c = 1, d = 1, we think every a=1, b=1, c=1 only filtrate 1/100 data,
+		// so as to avoid collapsing too fast.
+		for j := 0; j < i; j++ {
+			count = count / float64(100)
+		}
+		totalCount += count
+	}
+	// To avoid the totalCount become 0.
+	if uint64(totalCount) == 0 {
+		totalCount = 10.0
+	}
```

Comment:
This code is also used to line 388, it can extract into a function. And use the same comments it will be clearer.